sqlmesh/migrations/v0008_create_intervals_table.py

```diff
@@ -1,21 +1,25 @@
 """Create a dedicated table to store snapshot intervals."""
 from sqlglot import exp
 
+from sqlmesh.utils.migration import primary_key_text_type
+
 
 def migrate(state_sync):  # type: ignore
     engine_adapter = state_sync.engine_adapter
     intervals_table = "_intervals"
     if state_sync.schema:
         intervals_table = f"{state_sync.schema}.{intervals_table}"
 
+    pk_text_type = primary_key_text_type(engine_adapter.dialect)
+
     engine_adapter.create_state_table(
         intervals_table,
         {
             "id": exp.DataType.build("text"),
             "created_ts": exp.DataType.build("bigint"),
-            "name": exp.DataType.build("text"),
-            "identifier": exp.DataType.build("text"),
-            "version": exp.DataType.build("text"),
+            "name": exp.DataType.build(pk_text_type),
+            "identifier": exp.DataType.build(pk_text_type),
+            "version": exp.DataType.build(pk_text_type),
```

Comment:
Please ignore this if I'm mistaken, but isn't the primary key for this table `id` and so `id`'s type needs to be updated but not these columns?

Reply:
everything that has an index needs to be varchar fixed

Reply:
Thank you for clarifying. So does `id` also need to have its type updated to varchar?

Reply:
the problem is that mysql doesn't allow unbounded chars in indices

Reply:
Sorry, I didn't mean just varchar, but the varchar(1024) in the pk_text_type variable. This create table will fail in MySQL if the `id` column is type text and it's used in the primary key index, right?

Reply:
yes @GeorgeSittas you need to update "id"

Reply:
You're right @vchan, thanks for catching this!